subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -462,13 +469,43 @@ uint8_t ull_scan_disable(uint8_t handle, struct ll_scan_set *scan)
 {
 	int err;
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	/* Request Extended Scan stop */
+	scan->is_stop = 1U;
+	cpu_dmb();
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
+
 	err = ull_ticker_stop_with_mark(TICKER_ID_SCAN_BASE + handle,
 					scan, &scan->lll);
 	LL_ASSERT(err == 0 || err == -EALREADY);
 	if (err) {
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	/* Find and stop associated auxiliary scan contexts */
+	for (uint8_t aux_handle = 0; aux_handle < CONFIG_BT_CTLR_SCAN_AUX_SET;
+	     aux_handle++) {
+		struct lll_scan_aux *aux_scan_lll;
+		struct ll_scan_set *aux_scan;
+		struct ll_scan_aux_set *aux;
+
+		aux = ull_scan_aux_set_get(aux_handle);
+		aux_scan_lll = aux->parent;
+		if (!aux_scan_lll) {
+			continue;
+		}
+
+		aux_scan = HDR_LLL2ULL(aux_scan_lll);
+		if (aux_scan == scan) {
+			err = ull_scan_aux_stop(aux);
+			if (err && (err != -EALREADY)) {
+				return BT_HCI_ERR_CMD_DISALLOWED;
```

Comment:
This conditional should not occur, should I replace this with a LL_ASSERT?

Reply:
If that is something that may not happen and it will get to some unexpected state then in my humble opinion we should assert. I would not create an assert if there is a way to recover from the situation. The change can be added in follow up PR.